src/BugsnagUnity/Delivery.cs

```diff
@@ -95,24 +98,164 @@ public void Deliver(IPayload payload)
             }
         }
 
-        private byte[] TruncateBreadcrumbs(IPayload payload, byte[] serialisedPayload)
+        // Push to the server and handle the result
+        IEnumerator PushToServer(IPayload payload)
+        {
+            var shouldDeliver = false;
+            if (Application.platform == RuntimePlatform.WebGLPlayer)
+            {
+                shouldDeliver = _client.NativeClient.ShouldAttemptDelivery();
+            }
+            else
+            {
+                var networkCheckDone = false;
+                new Thread(() => {
+                    shouldDeliver = _client.NativeClient.ShouldAttemptDelivery();
+                    networkCheckDone = true;
+                }).Start();
+
+                while (!networkCheckDone)
+                {
+                    yield return null;
+                }
+            }
+            if (!shouldDeliver)
+            {
+                _payloadManager.SendPayloadFailed(payload);
+                _finishedCacheDeliveries.Add(payload.Id);
+                yield break;
+            }
+
+            byte[] body = new byte[] { };
```

Comment:
This initialised value is never used is it?

Reply:
the compiler was complaining when i set it to null before but now it's fine. Must be some change to the if afterwards or something. Set to null now 👍

Reply:
tested this and it fails silently if null, so no need for a guard

Reply:
in a later ticket we will handle errors in delivery in a better way